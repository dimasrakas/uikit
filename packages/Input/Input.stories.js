import Input from "./Input.vue";

// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
export default {
  title: "Design System/Input",
  component: Input,
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { Input },
  template: '<Input v-bind="$props" />',
});

export const TextInput = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
TextInput.args = {
  id: "text-input",
  placeholder: "Your placeholder",
};
