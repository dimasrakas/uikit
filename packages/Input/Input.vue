<template>
  <div>
    <label :for="id"> {{ label }} </label>
    <input :id="id" type="text" :placeholder="placeholder" />

    {{ state }}
  </div>
</template>

<script>
export default {
  props: {
    id: {
      type: [String],
      default: "text",
    },
    label: {
      type: [String],
      default: "Label name",
    },
    placeholder: {
      type: [String],
      default: "Placeholder",
    },
    state: {
      type: [String],
      default: "State",
    },
  },
};
</script>

<style lang="scss" scoped></style>
