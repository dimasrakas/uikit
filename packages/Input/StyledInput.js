import styled from "vue-styled-components";
import { space } from "styled-system";

import { colors, space as _space } from "@dimasrakas/themes";

const inputProps = {
  error: Boolean,
  px: { default: _space[3], type: [Number, String, Array] },
  py: { default: _space[2], type: [Number, String, Array] },
};

const StyledInput = styled("input", inputProps)`
  font-size: 100%;
  color: ${colors.gray[900]};
  background: ${colors.white};
  border: 1px solid
    ${(props) => (props.error ? colors.red[500] : colors.gray[200])};
  border-radius: 8px;

  &:focus {
    outline: 2px solid ${colors.blue[600]};
  }

  ::placeholder,
  ::-webkit-input-placeholder {
    color: ${colors.gray[300]};
  }
  :-ms-input-placeholder {
    color: ${colors.gray[300]};
  }

  ${space}
`;

export default StyledInput;
