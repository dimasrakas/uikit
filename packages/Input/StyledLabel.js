import styled from "vue-styled-components";
import { colors, fontSize, space } from "@dimasrakas/themes";

export default styled("label")`
  display: block;
  margin-bottom: 0.5rem;
  font-weight: medium;
  font-size: ${fontSize.base};
  color: ${colors.gray[800]};
`;
