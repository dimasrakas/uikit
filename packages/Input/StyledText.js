import styled from "vue-styled-components";
import { space, fontWeight, lineHeight, fontSize, color } from "styled-system";

export default styled("span", {
  fontSize: [Number, Array],
  fontWeight: String,
  color: String,
})`
  line-height: none;
  display: block;
  margin-top: 4px;
  font-size: 0.75rem;
  color: gray;
  ${space}
  ${fontSize}
  ${fontWeight}
  ${lineHeight}
  ${color}
`;
