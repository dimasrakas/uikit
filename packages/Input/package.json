{
  "name": "@dimasrakas/input",
  "version": "1.0.1-alpha.0",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "styled-system": "4.2.4",
    "vue-styled-components": "1.6.0",
    "@dimasrakas/themes": "^1.0.1-alpha.0"
  }
}
