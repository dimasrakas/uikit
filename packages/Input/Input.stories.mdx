import { Meta, Story, Canvas } from "@storybook/addon-docs/blocks";
import { ArgsTable } from "@storybook/addon-docs";

import Input from "./Input.vue";

<Meta title="Design System/Input" component={Input} />

export const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { Input },
  template: '<Input v-bind="$props" />',
});

# Input

With `MDX`, we can define a story for `Button` right in the middle of our
Markdown documentation.

## Install

```bash
yarn add @dimasrakas/input

// or

npm install @dimasrakas/input
```

## Imports

```js
import { Input } from "@dimasrakas/input";
```

## Usage

### Text Input

<Canvas withSource="open">
  <Story
    name="Text"
    args={{
      type: "text",
      id: "text-input",
      placeholder: "Your full name",
      label: "My name",
      infoMessage: "Tip: Simple caption text here",
      errorMessage: "",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

### Number Input

<Canvas withSource="open">
  <Story
    name="Number"
    args={{
      type: "number",
      id: "text-input",
      placeholder: "Your number",
      label: "My age",
      infoMessage: "Tip: Simple caption text here",
      errorMessage: "",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

### Date Input

<Canvas withSource="open">
  <Story
    name="Date"
    args={{
      type: "date",
      id: "text-input",
      placeholder: "Your date",
      label: "My birthday",
      infoMessage: "Tip: Simple caption text here",
      errorMessage: "",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

### Password Input

<Canvas withSource="open">
  <Story
    name="Password"
    args={{
      type: "password",
      id: "text-input",
      placeholder: "Your password",
      label: "My password",
      infoMessage: "Tip: Simple caption text here",
      errorMessage: "",
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

## Props

<ArgsTable of={Input} />
