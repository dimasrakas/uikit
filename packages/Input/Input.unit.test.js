import { shallowMount } from "@vue/test-utils";
import Input from "./Input.vue";

describe("Input.vue", () => {
  it('should have "Input" as name', () => {
    expect(Input.name).toBe("Input");
  });

  it("should render correctly", () => {
    const wrapper = shallowMount(Input);
    expect(wrapper.element).toMatchSnapshot();
  });
});
