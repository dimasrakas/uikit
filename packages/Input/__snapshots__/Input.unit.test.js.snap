// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.vue should render correctly 1`] = `
<div>
  <!---->
   
  <styledinput-stub
    id="text"
    placeholder="Placeholder"
    px="3"
    py="2"
    type="text"
  />
   
  <!---->
   
  <!---->
</div>
`;
